docs/query-builder.md

```diff
@@ -448,64 +448,161 @@ DB::collection('items')
 
 **Push**
 
-Add items to an array.
+Add one or multiple values to the items array.
 
 ```php
+// Push the value to the matched documents
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots');
+  ->where('name', 'John')
+  // Push a single value to the items array
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  // Push multiple values to the items array
+  ->push('items', ['hat', 'jeans']);
+// Result:
+// items: ['boots', 'hat', 'jeans']
 
+// Or
+
+// Push the value to an exact document
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add an array as a value to the messages array. value arrays must be nested into a list array, otherwise they are
+interpreted as a list of scalar values.
```

Comment:
```suggestion
To add embedded document or array values to the `messages` array, those values must be specified within a list array.
```

> otherwise they are interpreted as a list of scalar values.

I think this raises more questions, as it's unclear what would happen if the second argument to `push()` was an associative array. I'd hope there'd be an exception thrown if the argument is not a real list, but I wouldn't be surprised if it was undefined behavior (maybe keys are ignored).

Offhand, does the method internally wrap its arguments in an [`$each`](https://www.mongodb.com/docs/manual/reference/operator/update/each/#mongodb-update-up.-each) operator?

Reply:
https://github.com/mongodb/laravel-mongodb/blob/973b6b90f7916916bf252ebd707fe4f4b5eaaeb1/src/Eloquent/Model.php#L470

Yes, It does.